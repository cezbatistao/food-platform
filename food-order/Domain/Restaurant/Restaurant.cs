using System.Collections.Generic;

namespace food_order.Domain.Restaurant
{
    public class Restaurant
    {
        public Restaurant(long id, string uuid, string name, 
            string address, List<MenuItem> itens)
        {
            Id = id;
            Uuid = uuid;
            Name = name;
            Address = address;
            Itens = itens;
        }

        public long Id { get; }
        public string Uuid { get; }
        public string Name { get; }
        public string Address { get; }
        public List<MenuItem> Itens { get; }
    }
}
