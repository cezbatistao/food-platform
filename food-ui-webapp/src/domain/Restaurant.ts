export default class Restaurant {
  _id: string;
  _name: string;
  _logo: string;
  _description: string;
  _address: string;

  constructor(id: string, name: string, logo: string, description: string, 
              address: string) {
    this._id = id;
    this._name = name;
    this._logo = logo;
    this._description = description;
    this._address = address;
  }

  get id(): string {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get logo(): string {
    return this._logo;
  }

  get description(): string {
    return this._description;
  }

  get address(): string {
    return this._address;
  }
}