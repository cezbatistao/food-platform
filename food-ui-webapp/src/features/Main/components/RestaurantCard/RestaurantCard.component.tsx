import { useHistory } from 'react-router-dom';

type Props = {
  uuid: string,
  name: string,
  logo: string, 
  description: string,
  address: string
}

const RestaurantCard = ({uuid, name, logo, description, address}: Props) => {

  let history = useHistory();

  const handleClickCard = (uuidRestaurant: string) => {
    history.push(`/order/${uuidRestaurant}`);
  }

  return (
    <div 
      key={ uuid } 
      onClick={ () => handleClickCard(uuid) } 
      className="col"
      style={{cursor:'pointer'}} 
    >
      <div data-test="card_header_restaurant_data" className="card h-100">
        <div className="card-header bg-transparent"><h5>{ name }</h5></div>
        <img src={ logo } className="card-img-top" alt={ name } />
        <div className="card-body">
          <h6 className="card-title">{ address }</h6>
          <p className="card-text"></p>
          <div className="card-footer bg-transparent">TODO: Add stars from rating here!</div>
        </div>
      </div>
    </div>
  );
}

export default RestaurantCard;