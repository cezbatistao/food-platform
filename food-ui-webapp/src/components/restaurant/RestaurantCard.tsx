import { makeStyles } from '@material-ui/core/styles';
import { red } from '@material-ui/core/colors';

import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import CardHeader from '@material-ui/core/CardHeader';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import IconButton from '@material-ui/core/IconButton';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import CardContent from '@material-ui/core/CardContent';

import { useSelector } from 'react-redux';

import Restaurant from '../../domain/Restaurant';
import { RootState } from '../../gateway';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

type Props = {
  name: string,
  logo: string, 
  description: string,
  address: string
}

type RestaurantState = {
  restaurants: Restaurant[], 
  loading: boolean
}

const RestaurantCard = ({name, logo, description, address}: Props) => {

  const classes = useStyles();

  const { loading, restaurants }: RestaurantState = useSelector(
    (state: RootState) => state.restaurant
  );

  return (
    <div>
      <Card className={classes.root}>
        <CardHeader
          avatar={
            <Avatar aria-label="recipe" className={classes.avatar}>
              PH
            </Avatar>
          }
          action={
            <IconButton aria-label="settings">
              <MoreVertIcon />
            </IconButton>
          }
          title={ name }
          subheader={ address }
        />
        <CardMedia
          className={classes.media}
          image={ logo }
          title={ name }
        />
        <CardContent>
          <Typography variant="body2" color="textSecondary" component="p">
            { description }
          </Typography>
        </CardContent>
      </Card>
    </div>
  );
}

export default RestaurantCard;