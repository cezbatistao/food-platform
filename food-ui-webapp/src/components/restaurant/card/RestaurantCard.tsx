import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';

import Restaurant from '../../../domain/Restaurant';
import { RootState } from '../../../gateway';

type Props = {
  uuid: string,
  name: string,
  logo: string, 
  description: string,
  address: string
}

const RestaurantCard = ({uuid, name, logo, description, address}: Props) => {

  let history = useHistory();

  const handleClickCard = (uuidRestaurant: string) => {
    history.push(`/order/${uuidRestaurant}`);
  }

  return (
    <div 
      onClick={ () => handleClickCard(uuid) } 
      key={ uuid } 
      className="col"
    >
      <div className="card h-100">
        <div className="card-header bg-transparent"><h5>{ name }</h5></div>
        <img src={ logo } className="card-img-top" alt={ name } />
        <div className="card-body">
          <h6 className="card-title">{ address }</h6>
          <p className="card-text"></p>
          <div className="card-footer bg-transparent">TODO: Add stars from rating here!</div>
          <a href="#" className="stretched-link"></a>
        </div>
      </div>
    </div>
  );
}

export default RestaurantCard;