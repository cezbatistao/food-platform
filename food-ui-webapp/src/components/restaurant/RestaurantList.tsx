import { useSelector } from 'react-redux';

import Grid from '@material-ui/core/Grid';

import RestaurantCard from './RestaurantCard';

import { RootState } from '../../gateway';
import Restaurant from '../../domain/Restaurant';

type RestaurantState = {
  restaurants: Restaurant[], 
  loading: boolean
}

const RestaurantList = () => {

  const { loading, restaurants }: RestaurantState = useSelector((
    state: RootState
  ) => state.restaurant);

  return (
    <Grid container spacing={3}>
      {
        restaurants.map(restaurant => {
          return(
            <Grid key={ restaurant.id } item xs={12} sm={4}>
              <RestaurantCard 
                name={ restaurant.name } 
                logo={ restaurant.logo }
                description={ restaurant.description }
                address={ restaurant.address }
              />
            </Grid>
          );
        })
      }
    </Grid>
  );
}

export default RestaurantList;