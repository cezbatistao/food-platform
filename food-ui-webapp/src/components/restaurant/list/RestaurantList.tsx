import { Fragment } from 'react';
import { useSelector } from 'react-redux';

import RestaurantCard from '../card/RestaurantCard';
import { RootState } from '../../../gateway';
import Restaurant from '../../../domain/Restaurant';

type RestaurantState = {
  restaurants: Restaurant[], 
  loading: boolean
}

const RestaurantList = () => {

  const { loading, restaurants }: RestaurantState = useSelector((
    state: RootState
  ) => state.restaurant);
  
  return (
    <div className="row row-cols-1 row-cols-md-5 g-4">
      {
        restaurants.map(restaurant => {
          return(
            <RestaurantCard 
              uuid={ restaurant.uuid } 
              name={ restaurant.name } 
              logo={ restaurant.logo }
              description={ restaurant.description }
              address={ restaurant.address }
            />
          );
        })
      }
    </div>
  );
}

export default RestaurantList;