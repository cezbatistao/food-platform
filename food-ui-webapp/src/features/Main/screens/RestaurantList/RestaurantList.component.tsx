import { useState, useEffect, useContext } from "react";

import CategoryContext from '../../contexts/RestaurantCategory/RestaurantCategory.context';

import { fetchRestaurantsByCategory } from "../../../../services/Restaurant.services";
import RestaurantCard from "../../components/RestaurantCard/RestaurantCard.component";
import Restaurant from "../../../../services/Restaurant.model";

const RestaurantList = () => {

  const { category } = useContext(CategoryContext);

  const [restaurants, setRestaurants] = useState<Restaurant[]>([]);
  const [loading, setLoading] = useState<boolean>(true);

  useEffect(() => {
    const getRestaurantsFromCategory = async () => {
      if(category) {
        const restaurantsFromCategory: Restaurant[] = await fetchRestaurantsByCategory(category);
        setRestaurants(restaurantsFromCategory);
        setLoading(false);
      }
    }

    setLoading(true);
    getRestaurantsFromCategory();
  }, [category]);

  return (
    <div key="restaurant-list" className="row row-cols-1 row-cols-md-5 g-4">
      {!loading &&
        restaurants.map(restaurant => {
          return(
            <RestaurantCard 
              key={ restaurant.uuid } 
              uuid={ restaurant.uuid } 
              name={ restaurant.name } 
              logo={ restaurant.logo }
              description={ restaurant.description }
              address={ restaurant.address }
            />
          );
        })
      }
    </div>
  );
}
  
export default RestaurantList;