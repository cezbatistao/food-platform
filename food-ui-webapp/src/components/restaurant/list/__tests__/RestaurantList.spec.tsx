import { shallow } from 'enzyme';

import { useSelector } from 'react-redux';

import RestaurantList from "../RestaurantList";
import RestaurantCard from '../../card/RestaurantCard';

import Restaurant from '../../../../domain/Restaurant';

import restaurantsResponseJson from "../../../../../dependencies/stubby/mocks/response/restaurants_response.json";

const mockUseSelector: jest.Mock = useSelector as any;
const mockUseDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockUseDispatch
}));

describe('RestaurantList', () => {

  beforeEach(() => {
    mockUseDispatch.mockClear();
    mockUseSelector.mockClear();
  });

  it("should don't show RestaurantCard", () => {
    mockUseSelector.mockImplementation((selector) => selector({
      restaurant: {
        restaurants: [], 
        error: false, 
        loading: false,
      }
    }));

    const wrapper = shallow(<RestaurantList />)
  
    expect(wrapper.find(RestaurantCard).length).toBe(0);

    expect(wrapper).toMatchSnapshot();
  });

  it("should show RestaurantCard", () => {
    mockUseSelector.mockImplementation((selector) => selector({
      restaurant: {
        restaurants: restaurantsResponseJson.data as Restaurant[], 
        error: false, 
        loading: false,
      }
    }));

    const wrapper = shallow(<RestaurantList />);
  
    expect(wrapper.find(RestaurantCard).length).toBe(3);

    expect(wrapper).toMatchSnapshot();
  });

});