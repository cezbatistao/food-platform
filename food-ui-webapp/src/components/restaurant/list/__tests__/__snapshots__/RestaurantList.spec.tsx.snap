// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList should don't show RestaurantCard 1`] = `
<div
  className="row row-cols-1 row-cols-md-3 g-4"
/>
`;

exports[`RestaurantList should show RestaurantCard 1`] = `
<div
  className="row row-cols-1 row-cols-md-3 g-4"
>
  <RestaurantCard
    address="Av. Nome da avenida, 123"
    description="Pizza Hut é uma cadeia de restaurantes e franquias especializada em pizzas e massas. Com sede na cidade de Plano, no Texas, a Pizza Hut é a maior cadeia de pizzarias do mundo, com quase 15 mil restaurantes e quiosques em mais de 130 países. Possui 95 restaurantes no Brasil e 91 em Portugal."
    logo="https://pbs.twimg.com/profile_images/1333417326704791553/Mm0bj3oN.jpg"
    name="Pizza Hut"
    uuid="cbb9c2bd-abde-48a3-891a-6229fc9b7c2f"
  />
  <RestaurantCard
    address="Av. Nome da outra avenida, 789"
    description="Pizzaria reúne ingredientes clássicos e sofisticados e muito burburinho e aconchego frente aos fornos à lenha."
    logo="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSwwg9Nscz6dkUtIpPUB0u1tAxfZmJcmcfGTA&usqp=CAU"
    name="Bráz Pizzaria"
    uuid="23e0211c-19ea-47ee-b98e-77e023e1a95f"
  />
  <RestaurantCard
    address="Rua Nome da rua, 654"
    description="Domino's Pizza é uma empresa de alimentação baseada em pizzas. Atualmente, é a maior rede de entregas de pizzas do mundo, com 13.000 lojas em 83 países"
    logo="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQd7Ow6kgZoZk4h4oybgOkRJf6LZ_NLpBhfRA&usqp=CAU"
    name="Domino's Pizza"
    uuid="36159a9b-f4d0-4f52-8d0f-3cd0dc702c1c"
  />
</div>
`;
